components/Button.tsx

```diff
@@ -1,25 +1,47 @@
 import * as React from 'react'
 
 type Props = {
-  color: string
+  color?: string
   text: string
   initial?: string | undefined
 }
 
 const Button = ({ color, initial, text }: Props) => {
+  let btnClass
+  switch (color) {
```

Comment:
To make this extensible, I think `type` is a better prop name. Then you can generate class like this:

```
const btnClass = `btn-${type}`
```